docs/design/pattern_matching.md

```diff
@@ -685,13 +690,32 @@ In order to match a value, whatever is specified in the pattern must match.
 Using `auto` for a type will always match, making `_: auto` the wildcard
 pattern.
 
+#### Guards
```

Comment:
Do we have an approved design for `match` statements and their syntax? This would seem to fit better in a separate documentation file specifically describing `match` statements, `case`, `default`, etc. than here, if we're ready to write such a file.

Reply:
Let's defer that for now.